### PR TITLE
Merge pull request #1 from hyperledger/master

### PR DESCRIPTION
**Pull request by silliman**
Update from original

### CODE REVIEW

**Comment by silliman** [comment]
I did not mean to merge from my fork into the main branch.  I wanted to refresh my fork from the main branch.  Please close this request and I'm sorry for the inconvenience.

File: docs/source/hsm.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+## Setting up a network using HSM
+
+If you are deploying Fabric nodes using an HSM, your private keys need to be
+generated inside the HSM rather than inside the `keystore` folder of the node's
+local MSP folder. The `keystore` folder of the MSP will remain empty. Instead,
+the Fabric node will use the subject key identifier of the signing certificate
+in the `signcerts` folder to retrieve the private key from inside the HSM.
+The process for creating the MSP folders will be different depending on if you
+are using a Fabric Certificate Authority (CA) your own CA.
```
Comment (denyeart) [inline docs/source/hsm.md]: typo - "if you are using a Fabric Certificate Authority (CA) your own CA."

File: docs/source/hsm.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+3. Edit the Fabric CA client config file or environment variables to use your
+HSM as the crypto service provider. Then for each node, use the Fabric CA client
+to generate the component MSP folder by enrolling against the node identity. The
+enroll command will generate the private key inside your HSM.
+
+3. Update the BCCSP section of the peer or orderer configuration file to use
+PKCS11 and your token as the crypto service provider. Point to the MSP that was
+generated using the Fabric CA client. Once it is deployed, the peer or orderer
+node will be able sign and endorse transactions with the private key protected by
+the HSM.
```
Comment (denyeart) [inline docs/source/hsm.md:140]: I'm not following this part... how do you get the HSM protected private key generated by Fabric CA client over to the peer/orderer HSM? What does "Once it is deployed mean"?

File: docs/source/hsm.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+You can also use environment variables to override the relevant fields of the
+configuration file. If you are connecting to an HSM using the Fabric CA server,
+you need to set the following environment variables:
+
+```
+FABRIC_CA_SERVER_BCCSP_DEFAULT=PKCS11
+FABRIC_CA_SERVER_BCCSP_PKCS11_LIBRARY=/etc/hyperledger/fabric/libsofthsm2.so
+FABRIC_CA_SERVER_BCCSP_PKCS11_PIN=71811222
+FABRIC_CA_SERVER_BCCSP_PKCS11_LABEL=fabric
```
Comment (denyeart) [inline docs/source/hsm.md:90]: Is there a reason the CA environment variables are listed, but not the peer/orderer environment variables?

**Pull request closed without merging**